### ISSUE TITLE
Training Loss

### ISSUE THREAD

**Issue by ZongyangLi** [junyanz/pytorch-CycleGAN-and-pix2pix#303]
I am using pix2pix to train a gray2rgb model, the output from the model seems reasonable to me, but not good enough, but I am confuse about the plot of loss, GAN_LOSS continues to increase, in the same time D_REAL_LOSS and D_FAKE_LOSS is really low. Could anyone tell me what might be the problem? Training image is about 30,000 but they are all in a same style.

Plot of loss:
![newplot](https://user-images.githubusercontent.com/20230686/41724810-fb9befcc-7533-11e8-804a-8d3ce48a3d5a.png)

Example training image:
![2016-10-10__11-14-59-441_plot_0](https://user-images.githubusercontent.com/20230686/41724838-137833d0-7534-11e8-8f4c-8e6394ad6f2a.jpg)

Testing output:
![2016-09-13__12-55-58-192_plot_0_real_a](https://user-images.githubusercontent.com/20230686/41724905-3c45660c-7534-11e8-9c3a-559712cc96f5.png)
![2016-09-13__12-55-58-192_plot_0_real_b](https://user-images.githubusercontent.com/20230686/41724916-412a344a-7534-11e8-9871-c37a0a9e9dbe.png) **real**
![2016-09-13__12-55-58-192_plot_0_fake_b](https://user-images.githubusercontent.com/20230686/41724920-43828abc-7534-11e8-9073-6c55e08fce91.png) **fake**

**Comment by tangjilin** [comment]
Hello, I have the similar situation with you.Have you solved it? I have 10000 images to train my model.
My loss plot is like this:
![cg vm2uvov hyj 1ae z1](https://user-images.githubusercontent.com/35884085/42016803-73cb51e4-7adf-11e8-8ce7-aacbf97a6671.png)

**Comment by tangjilin** [comment]
Hello, I have the similar situation with you.Have you solved it? I have 10000 images to train my model.
My loss plot is like this:
![my](https://user-images.githubusercontent.com/35884085/42017108-80022cca-7ae0-11e8-83d1-dfeb51a46ce7.png)

**Comment by ZongyangLi** [comment]
@tangjilin I am guessing it's the right loss plot from previous issues. Something different between our plots is your G_L1 loss is higher, and my G_GAN loss is higher. May I ask you what kind of operation did you apply to your input images?

**Comment by tangjilin** [comment]
I stack the A and B along the channel to as my input.

发自我的 iPhone

在 2018年7月2日，下午11:16，ZongyangLi <notifications@github.com<mailto:notifications@github.com>> 写道：


@tangjilin<https://github.com/tangjilin> I am guessing it's the right loss plot from previous issues. Something different between our plots is your G_L1 loss is higher, and my G_GAN loss is higher. May I ask you what kind of operation did you apply to your input images?

**Comment by junyanz** [comment]
I think it's normal. `G_GAN` is the GAN loss for the generator. `D_real` and `D_fake` are the losses for Ds regarding real and fake inputs.

**Comment by taesung89** [comment]
The plot looks normal to me. The loss curve doesn't usually reveal much information, so I cannot tell whether the training was successful or not just based on the loss curve.

**Comment by ranjiewwen** [comment]
while when i train D_real and D_fake convergence 0，is this mean training model collapse? normal D_real and D_fake will convergence 0.5, which can not distinguish real or fake.

**Comment by junyanz** [comment]
0 might means that D is overfitting the training set and winning the optimization game. 0.1-0.3 is a normal range.
